lib/cppcheck.cpp

```diff
@@ -1294,6 +1294,90 @@ void CppCheck::getErrorMessages()
     Preprocessor::getErrorMessages(this, &s);
 }
 
+bool CppCheck::analyseClangTidy(const ImportProject::FileSettings &fileSettings )
+{
+    std::string allIncludes = "";
+    std::string allDefines = "-D"+fileSettings.defines;
+    for (std::string inc : fileSettings.includePaths) {
+        allIncludes = allIncludes + "-I"" + inc + "" ";
+    }
+    allIncludes = allIncludes + "-I. ";
+
+    std::string::size_type pos = 0u;
+    while ((pos = allDefines.find(";", pos)) != std::string::npos)
+    {
+        allDefines.replace(pos, 1, " -D");
+        pos += 3;
+    }
+
+    const std::string cmd = "clang-tidy -quiet -checks=*,-clang-analyzer-*,-llvm* " + fileSettings.filename + " -- " + allIncludes + allDefines;
+
+    #ifdef _WIN32
+        std::unique_ptr<FILE, decltype(&_pclose)> pipe(_popen(cmd.c_str(), "r"), _pclose);
+    #else
+        std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"), pclose);
+    #endif
+    if (!pipe)
+        throw InternalError(nullptr, "popen failed (command: '" + cmd + "')");
+    char buffer[1024];
+    std::string result;
+    while (fgets(buffer, sizeof(buffer), pipe.get()) != nullptr)
+    {
+        result += buffer;
+    }
+
+    // parse output and create error messages
+    std::istringstream istr(result);
+    std::string line;
+
+    while (std::getline(istr, line)) {
+        if (line.find("error") != std::string::npos || line.find("warning") != std::string::npos) {
+            std::vector<std::string> lineParts;
+            splitString(line, lineParts, ' ');
```

Comment:
You can use this function:  https://github.com/danmar/cppcheck/blob/master/lib/cppcheck.cpp#L187